swabbie-edda/src/main/kotlin/com/netflix/spinnaker/swabbie/edda/providers/EddaLaunchConfigurationCacheProvider.kt

```diff
@@ -0,0 +1,105 @@
+package com.netflix.spinnaker.swabbie.edda.providers
+
+import com.netflix.spinnaker.config.EddaApiClient
+import com.netflix.spinnaker.swabbie.*
+import com.netflix.spinnaker.swabbie.aws.launchconfigurations.AmazonLaunchConfiguration
+import com.netflix.spinnaker.swabbie.model.WorkConfiguration
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Lazy
+import java.lang.IllegalArgumentException
+import java.time.Clock
+
+@Configuration
+open class EddaLaunchConfigurationCacheProvider(
```

Comment:
`EddaLaunchConfigurationCacheProvider` provides:
* A cache of all `AmazonLaunchConfigs` for all accounts by region
* A cached hashmap keyed on AMI's actively referenced by a launch config in any account (by region), where values are a set of `AmazonLaunchConfig` currently referencing the key.
* Updated every N minutes (currently 15min based on the `Cacheable` default)
* Provides a `lastUpdated` attribute

Example use:
`AmazonImageHandler` could inject` launchConfigurationCache: CachedViewProvider<AmazonLaunchConfigurationCache>` and call: `val amisUsedByLaunchConfigs = launchConfigurationCache.getRefdAmisForRegion(region)`, then filter candidates where  `amisUsedByLaunchConfigs.contains(amiId)`. If `launchConfigurationCache.getLastUpdated()` is older than an expected value (i.e. now-1h), abort evaluation.